pkg/kubelet/kubelet.go

```diff
@@ -1533,7 +1533,8 @@ func (kl *Kubelet) Run(updates <-chan kubetypes.PodUpdate) {
 // This operation writes all events that are dispatched in order to provide
 // the most accurate information possible about an error situation to aid debugging.
 // Callers should not write an event if this operation returns an error.
-func (kl *Kubelet) syncPod(ctx context.Context, updateType kubetypes.SyncPodType, pod, mirrorPod *v1.Pod, podStatus *kubecontainer.PodStatus) (isTerminal bool, err error) {
+func (kl *Kubelet) syncPod(_ context.Context, updateType kubetypes.SyncPodType, pod, mirrorPod *v1.Pod, podStatus *kubecontainer.PodStatus) (isTerminal bool, err error) {
+	ctx := context.Background()
```

Comment:
/hold

this looks _really_ suspicious... we're ignoring the passed-in context and replacing it with a non-terminating one? doesn't this defeat the purpose of propagating the context that this PR is trying to accomplish?

if the passed-in context is bad for some reason, it seems like we should resolve that issue first, rather than breaking the chain here

same comment on syncTerminatingPod and syncTerminatedPod

Reply:
Thats reasonable.  https://github.com/kubernetes/kubernetes/pull/113566 resolves the bad context (at least it is one way of doing so).  This PR does at least demonstrate where the issue was coming from.

Reply:
Discussed this offline with clayton and david porter.  The plan is to ignore the incoming context, and add a context.TODO() with a link to an issue for cancelation in the kubelet, which @smarterclayton will open.

Reply:
Added TODOs

Reply:
Nice. thanks!

Reply:
resolving.  Checked with liggitt offline